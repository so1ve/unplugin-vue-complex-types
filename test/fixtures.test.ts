import { join } from "node:path";

import VueBetterDefine from "@vue-macros/better-define/rollup";
import {
	RollupEsbuildPlugin,
	RollupJson,
	RollupNodeResolve,
	RollupVue,
	rollupBuild,
	testFixtures,
} from "@vue-macros/test-utils";
import { describe } from "vitest";

import VueComplexTypes from "../src/rollup";

describe("fixtures", async () => {
	await testFixtures(
		["__fixtures__/*.vue"],
		(_args, id) =>
			rollupBuild(id, [
				VueComplexTypes({
					tsconfigPath: join(__dirname, "__fixtures__", "tsconfig.json"),
				}),
				VueBetterDefine(),
				RollupVue(),
				RollupJson(),
				RollupNodeResolve(),
				RollupEsbuildPlugin({
					target: "esnext",
				}),
			]),
		{
			cwd: __dirname,
			promise: true,
		},
	);
});
