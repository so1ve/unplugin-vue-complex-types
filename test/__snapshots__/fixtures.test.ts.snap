// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > __fixtures__/basic.vue 1`] = `
"<script
	setup
	lang=\\"ts\\"
	generic=\\"T extends string, U extends (string | number)[], I\\"
>
import type { SomeInterface } from \\"./basic-ts\\";

defineProps<
	{
string: string
optionalString?: string
stringUnion: \\"a\\" | \\"b\\"
genericExtendsString: string
array: string[]
genericArray: (string | number)[]
genericCondition: \\"a\\" | 1
} & {
emptyInterface: EmptyInterface
conditionalNumber: number
interface: Interface
typeLiteral: TypeLiteral
} & {
\\"1-1\\": string
\\"2-1\\": string
}
>();
</script>
"
`;

exports[`fixtures > __fixtures__/core#8286.vue 1`] = `
"<script setup lang=\\"ts\\">
import { ButtonHTMLAttributes } from \\"vue\\";

interface MyButtonProps extends ButtonHTMLAttributes {}

defineProps<{
autofocus: Booleanish | undefined
disabled: Booleanish | undefined
form: string | undefined
formaction: string | undefined
formenctype: string | undefined
formmethod: string | undefined
formnovalidate: Booleanish | undefined
formtarget: string | undefined
name: string | undefined
type: \\"submit\\" | \\"reset\\" | \\"button\\" | undefined
value: string | number | string[] | undefined
innerHTML: string | undefined
class: any
style: StyleValue | undefined
accesskey: string | undefined
contenteditable: Booleanish | \\"inherit\\" | undefined
contextmenu: string | undefined
dir: string | undefined
draggable: Booleanish | undefined
hidden: Booleanish | undefined
id: string | undefined
lang: string | undefined
placeholder: string | undefined
spellcheck: Booleanish | undefined
tabindex: Numberish | undefined
title: string | undefined
translate: \\"yes\\" | \\"no\\" | undefined
radiogroup: string | undefined
role: string | undefined
about: string | undefined
datatype: string | undefined
inlist: any
prefix: string | undefined
property: string | undefined
resource: string | undefined
typeof: string | undefined
vocab: string | undefined
autocapitalize: string | undefined
autocorrect: string | undefined
autosave: string | undefined
color: string | undefined
itemprop: string | undefined
itemscope: Booleanish | undefined
itemtype: string | undefined
itemid: string | undefined
itemref: string | undefined
results: Numberish | undefined
security: string | undefined
unselectable: \\"on\\" | \\"off\\" | undefined
inputmode: \\"none\\" | \\"text\\" | \\"tel\\" | \\"url\\" | \\"email\\" | \\"numeric\\" | \\"decimal\\" | \\"search\\" | undefined
is: string | undefined
'aria-activedescendant': string | undefined
'aria-atomic': Booleanish | undefined
'aria-autocomplete': \\"none\\" | \\"inline\\" | \\"list\\" | \\"both\\" | undefined
'aria-busy': Booleanish | undefined
'aria-checked': Booleanish | \\"mixed\\" | undefined
'aria-colcount': Numberish | undefined
'aria-colindex': Numberish | undefined
'aria-colspan': Numberish | undefined
'aria-controls': string | undefined
'aria-current': Booleanish | \\"page\\" | \\"step\\" | \\"location\\" | \\"date\\" | \\"time\\" | undefined
'aria-describedby': string | undefined
'aria-details': string | undefined
'aria-disabled': Booleanish | undefined
'aria-dropeffect': \\"none\\" | \\"copy\\" | \\"execute\\" | \\"link\\" | \\"move\\" | \\"popup\\" | undefined
'aria-errormessage': string | undefined
'aria-expanded': Booleanish | undefined
'aria-flowto': string | undefined
'aria-grabbed': Booleanish | undefined
'aria-haspopup': Booleanish | \\"menu\\" | \\"listbox\\" | \\"tree\\" | \\"grid\\" | \\"dialog\\" | undefined
'aria-hidden': Booleanish | undefined
'aria-invalid': Booleanish | \\"grammar\\" | \\"spelling\\" | undefined
'aria-keyshortcuts': string | undefined
'aria-label': string | undefined
'aria-labelledby': string | undefined
'aria-level': Numberish | undefined
'aria-live': \\"off\\" | \\"assertive\\" | \\"polite\\" | undefined
'aria-modal': Booleanish | undefined
'aria-multiline': Booleanish | undefined
'aria-multiselectable': Booleanish | undefined
'aria-orientation': \\"horizontal\\" | \\"vertical\\" | undefined
'aria-owns': string | undefined
'aria-placeholder': string | undefined
'aria-posinset': Numberish | undefined
'aria-pressed': Booleanish | \\"mixed\\" | undefined
'aria-readonly': Booleanish | undefined
'aria-relevant': \\"text\\" | \\"additions\\" | \\"additions text\\" | \\"all\\" | \\"removals\\" | undefined
'aria-required': Booleanish | undefined
'aria-roledescription': string | undefined
'aria-rowcount': Numberish | undefined
'aria-rowindex': Numberish | undefined
'aria-rowspan': Numberish | undefined
'aria-selected': Booleanish | undefined
'aria-setsize': Numberish | undefined
'aria-sort': \\"none\\" | \\"ascending\\" | \\"descending\\" | \\"other\\" | undefined
'aria-valuemax': Numberish | undefined
'aria-valuemin': Numberish | undefined
'aria-valuenow': Numberish | undefined
'aria-valuetext': string | undefined
onCopy: any
onCut: any
onPaste: any
onCompositionend: any
onCompositionstart: any
onCompositionupdate: any
onDrag: any
onDragend: any
onDragenter: any
onDragexit: any
onDragleave: any
onDragover: any
onDragstart: any
onDrop: any
onFocus: any
onFocusin: any
onFocusout: any
onBlur: any
onChange: any
onBeforeinput: any
onInput: any
onReset: any
onSubmit: any
onInvalid: any
onLoad: any
onError: any
onKeydown: any
onKeypress: any
onKeyup: any
onAuxclick: any
onClick: any
onContextmenu: any
onDblclick: any
onMousedown: any
onMouseenter: any
onMouseleave: any
onMousemove: any
onMouseout: any
onMouseover: any
onMouseup: any
onAbort: any
onCanplay: any
onCanplaythrough: any
onDurationchange: any
onEmptied: any
onEncrypted: any
onEnded: any
onLoadeddata: any
onLoadedmetadata: any
onLoadstart: any
onPause: any
onPlay: any
onPlaying: any
onProgress: any
onRatechange: any
onSeeked: any
onSeeking: any
onStalled: any
onSuspend: any
onTimeupdate: any
onVolumechange: any
onWaiting: any
onSelect: any
onScroll: any
onTouchcancel: any
onTouchend: any
onTouchmove: any
onTouchstart: any
onPointerdown: any
onPointermove: any
onPointerup: any
onPointercancel: any
onPointerenter: any
onPointerleave: any
onPointerover: any
onPointerout: any
onWheel: any
onAnimationstart: any
onAnimationend: any
onAnimationiteration: any
onTransitionend: any
onTransitionstart: any
}>();
</script>
"
`;

exports[`fixtures > __fixtures__/import-from-vue.vue 1`] = `
"<script setup lang=\\"ts\\" generic=\\"T extends string\\">
import type { Foo } from \\"./import-from-vue-vue.exclude.vue\\";

defineProps<{
foo: string
}>();
</script>
"
`;
